src/Microsoft.AspNet.SignalR.Redis/RedisMessageBus.cs

```diff
@@ -27,6 +27,7 @@ public class RedisMessageBus : ScaleoutMessageBus
         private string _connectionString;
         private int _state;
         private readonly object _callbackLock = new object();
+        private readonly SemaphoreSlim _redisConnectionEventLock = new SemaphoreSlim(1, 1);
```

Comment:
Part of me is wondering now if we should switch to interlocked all over or lock all over. Mixing the 2 things seems strange.